### PR TITLE
Allow UDP transport for metrics

### PR DESCRIPTION
**Pull request by nmonterroso**
Adds support for sending data over UDP. This required a few key changes:
* Change the `Influxdb` class to be an interface, with two implementations: `InfluxdbHttp` and `InfluxdbUdp`
* Rework the way series data was appended to `JsonBuilder` objects. Rather than separate the two arguments `Object[] columns` and `Object[][] points`, combine them into a single `SeriesData` object. This was done so we could remove the `time` column from UDP requests, since that field is unsupported over UDP and any requests with that field fail the whole request.

/cc @drsnyder @davidB

### CODE REVIEW

**Comment by davidB** [comment]
Thanks for the PR, including the update of README (the only doc ;-) )

But I'm not a fan of the SeriesData abstraction. The purpose of the "reusable" array is to avoid allocation of temporary object. SeriesData create ArrayList, SeriesData, SeriesData.Builder,... (The performence and the memory impact of metrics collector have to be as small as possible (like for logging), because it can be called a lot of time.

Can you remove SeriesData (or give arguments to include it) ?

**Comment by nmonterroso** [comment]
Thanks for the PR lookover!

I added `SeriesData` because originally I thought that sending the `time` column over udp caused the metrics to be ignored by influxdb. Turns out, though, that my metrics were getting dropped because the time data is specified in milliseconds. According to https://github.com/influxdb/influxdb/issues/841, though, I need to specify my timestamps in seconds. I'll update the PR.

**Comment by nmonterroso** [comment]
@davidB - Re-added the reusable objects

File: pom.xml [hunk 1]
```diff
@@ -32,10 +32,9 @@
 	</licenses>
 	<dependencies>
 		<dependency>
-			<groupId>com.codahale.metrics</groupId>
+			<groupId>io.dropwizard.metrics</groupId>
 			<artifactId>metrics-core</artifactId>
-			<version>3.0.1</version>
-			<scope>provided</scope>
+			<version>3.1.0</version>
```
Comment (davidB) [inline pom.xml]: why do you remove the "provided" scope ?
I don't want to force the version and groupId of metrics.

**Comment by davidB** [comment]
Thanks for the correcttion, I'll merge it.
I don't know when I'll publish a release, i need to re-install influxdb + ...
But  I'll publish the snapshot this WE (I hope)

**Comment by nmonterroso** [comment]
Cool! Can you let me know when snapshots and release versions have been published? Looking forward to using this in an upcoming project :)

**Comment by davidB** [comment]
0.6.0-SNAPSHOT available in repository: 

 https://oss.sonatype.org/content/repositories/snapshots/

**Comment by nmonterroso** [comment]
Awesome! Are there any plans to sync to central?

**Comment by davidB** [comment]
Maybe next week.

On Wed, Jan 21, 2015 at 5:38 PM, Nelson Monterroso <notifications@github.com

**Comment by nmonterroso** [comment]
Ping.

**Comment by davidB** [comment]
Sorry, version 0.6.0 is send to maven central, it shoulds be available in 24H

**Pull request merged**